File: frontend/FrontApps/src/app/driver/model/driver.ts
import { Car } from "./car"

export class DriverResponse {
  public id: number
  public name: string
  public lastName: string
  public car: Car

  constructor(id: number, name: string, lastName: string, car: Car) {
    this.id = id
    this.name = name
    this.lastName = lastName
    this.car = car
  }
}

export class Driver {
  public name: string
  public lastName: string
  public carId: number
  public id?: number

  constructor(name: string, lastName: string, carId: number, id?: number) {
    this.name = name
    this.lastName = lastName
    this.carId = carId
    this.id = id
  }
}

File: backend/src/main/java/com/interactive/demo/model/Driver.java
package com.interactive.demo.model;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Driver {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String name;
    @Column(name = "last_name")
    private String lastName;

    @ManyToOne
    @JoinColumn(name = "car_id", nullable = false)
    private Car car;

    public Driver(String name, String lastName, Car car) {
        this.name = name;
        this.lastName = lastName;
        this.car = car;
    }
}